src/main/scala/tech/cryptonomic/conseil/generic/chain/DataTypes.scala

```diff
@@ -309,7 +312,7 @@ object DataTypes {
     /** Helper method for extracting prefixes needed for SQL */
     def prefixes: List[String] = values.toList.map(_.toString + "_")
     type AggregationType = Value
-    val sum, count, max, min, avg = Value
+    val sum, count, max, min, avg, datePart = Value
```

Comment:
Please provide a sample JSON query

Reply:
Sample query for `blocks` :
```
{
  "predicates": [
  ],
  "aggregation": [
    {
      "field": "timestamp",
      "function": "datePart",
      "format": "YYYY-MM-DD"
    },
    {
      "field": "hash",
      "function": "count",
    }
  ],
  "fields": [
    "timestamp", "hash"
  ],
  "orderBy": [
  ],
  "output": "json",
  "limit": 100
}
```

Reply:
This is potentially problematic. Aggregations can only be done on fields that are explicitly included in the fields section. That is probably the better place to put the `format`, because I could see the user wanting to format a date in a particular way even if they are not aggregating.

Reply:
so what you are suggesting is to have `"fields": [ {"name": "timestamp", "format":"YYYY-MM-DD"} ]` query, right? 
Important thing to note is that changing `fields` list won't be backward compatible.

Reply:
to be clear can you write json schema description for this case?

Reply:
I feel pretty strongly about this. the formatted field is being aggregated on, it's not the aggregate. The aggregation structure only contains aggregates, sums, counts, etc. The query schema should be as follows

```
{
  "fields": [
    "hash",
    { "field": "timestamp", "function": "datePart", "format": "YYYY-MM-DD" },
  ]
  "predicates": [ ... ],
  "aggregation": [
    { "field": "hash", "function": "count", }
  ],
  "orderBy": [ ... ],
  "output": "json",
  "limit": 100
}
```

I appreciate that this is potentially a breaking change, we can have a discussion about how this gets implemented so we can support both simple string and object elements inside the `fields` array.

Reply:
I am with @anonymoussprocket on this. DatePart is not a function with aggregation semantics, however it can hold `group by` semantics. 

As such, without looking at the history of what motivated this PR and hence risking and in due course being guilty of missing context, I would say it should not be part of the aggregation block in a query definition.

Reply:
Done

Reply:
shouldn't datePart be removed from that enum then?

Reply:
yes, I removed it, I've created another enum - for now it has one element